src/engine/engineworker.cpp

```diff
@@ -15,13 +16,21 @@ void EngineWorker::run() {
 }
 
 void EngineWorker::setScheduler(EngineWorkerScheduler* pScheduler) {
+    DEBUG_ASSERT(m_pScheduler == nullptr);
     m_pScheduler = pScheduler;
+    pScheduler->addWorker(this);
 }
 
-bool EngineWorker::workReady() {
-    if (m_pScheduler) {
-        m_pScheduler->workerReady(this);
-        return true;
+void EngineWorker::workReady() {
+    m_notReady.clear();
+    VERIFY_OR_DEBUG_ASSERT(m_pScheduler) {
+        return;     
+    }
+    m_pScheduler->workerReady();
+}
+
+void EngineWorker::wakeIfReady() {
+    if (!m_notReady.test_and_set()) {
```

Comment:
Damn. I assumed that clear() returns the previous state as a result.
Does it get easier to read if we rename m_notReady to m_idle? On the other hand this would introduce another term around "ready", so let's better leave it as it is.

Reply:
I had the same thought, and notReady is finally the least hurting name.